saleor/graphql/account/types.py

```diff
@@ -430,15 +451,29 @@ def _resolve_orders(orders):
         return OrdersByUserLoader(info.context).load(root.id).then(_resolve_orders)
 
     @staticmethod
-    def resolve_avatar(root: models.User, info, size=None):
-        if root.avatar:
-            return Image.get_adjusted(
-                image=root.avatar,
-                alt=None,
-                size=size,
-                rendition_key_set="user_avatars",
-                info=info,
+    def resolve_avatar(root: models.User, info, size=None, format=None):
+        if not root.avatar:
+            return
+
+        if not size:
+            return Image(url=root.avatar.url, alt=None)
+
+        format = format.lower() if format else None
+        size = get_thumbnail_size(size)
+
+        def _resolve_avatar(thumbnail):
+            url = (
+                prepare_image_proxy_url(root.id, "User", size, format)
+                if thumbnail is None
+                else thumbnail.image.url
```

Comment:
Maybe we could have a function that encapsulates exactly this - returns proxy URL or thumbnail URL it's ready?